src/client/datascience/kernel-launcher/kernelLauncher.ts

```diff
@@ -99,17 +126,34 @@ export class KernelLauncher implements IKernelLauncher {
             this.extensionChecker,
             this.kernelEnvVarsService
         );
-        await kernelProcess.launch(workingDirectory);
+        await kernelProcess.launch(workingDirectory, timeout, cancelToken);
+
+        // Double check for cancel
+        if (cancelToken?.isCancellationRequested) {
```

Comment:
Is this required here.
I would expect cancellation to be checked before we do somethign that CPU intensive of the like.
I.e. if something has already happened and user requested a caancellation, i don't see the need to throw an error.
I'd expect a higher level to handle that scenario.

Reply:
Nothing to fix here.